Panic at enum compilation

This test of enum with `iota` passed in neo-go v0.90.0 but failed in v0.91.0

```go
	{
		"iota",
		`
		package foo
		type enum int
		const (
			_ enum = iota
			one
			two
        	)
		func Main() int {
			return int(two)
		}
		`,
		big.NewInt(2),
	},
```

Compiler throws panic:

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 11 [running]:
testing.tRunner.func1.1(0xb39e20, 0xc000026f20)
	/snap/go/current/src/testing/testing.go:988 +0x30d
testing.tRunner.func1(0xc000194ea0)
	/snap/go/current/src/testing/testing.go:991 +0x3f9
panic(0xb39e20, 0xc000026f20)
	/snap/go/current/src/runtime/panic.go:969 +0x166
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0xc000195320, 0xda6a00, 0xc0000211c0, 0x0, 0x0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/codegen.go:411 +0x8128
go/ast.Walk(0xda1420, 0xc000195320, 0xda6a00, 0xc0000211c0)
	/snap/go/current/src/go/ast/walk.go:52 +0x66
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).convertGlobals.func1(0xda6a00, 0xc0000211c0, 0xda6f01)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/codegen.go:281 +0x9e
go/ast.inspector.Visit(0xc00005f330, 0xda6a00, 0xc0000211c0, 0xda2520, 0xc00005f330)
	/snap/go/current/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0xda2520, 0xc00005f330, 0xda6a00, 0xc0000211c0)
	/snap/go/current/src/go/ast/walk.go:52 +0x66
go/ast.walkDeclList(0xda2520, 0xc00005f330, 0xc000021280, 0x3, 0x4)
	/snap/go/current/src/go/ast/walk.go:38 +0x9e
go/ast.Walk(0xda2520, 0xc00005f330, 0xda68c0, 0xc00018a580)
	/snap/go/current/src/go/ast/walk.go:353 +0x264e
go/ast.Inspect(...)
	/snap/go/current/src/go/ast/walk.go:385
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).convertGlobals(0xc000195320, 0xc00018a580, 0xc00007ceb0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/codegen.go:276 +0x7d
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).traverseGlobals.func2(0xc00018e580)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/analysis.go:72 +0xa5
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).ForEachPackage(0xc000195320, 0xc0001c9c20)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/compiler.go:60 +0x44
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).traverseGlobals(0xc000195320, 0xc0001c9cd0, 0x120)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/analysis.go:68 +0xf7
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).compile(0xc000195320, 0xc0001be5c0, 0xc00018e580, 0xc00018e580, 0x0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/codegen.go:1559 +0xde
github.com/nspcc-dev/neo-go/pkg/compiler.CodeGen(0xc0001be5c0, 0x6, 0xb41820, 0xc0001be280, 0xc0001be5c0, 0x0, 0x0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/codegen.go:1612 +0x88
github.com/nspcc-dev/neo-go/pkg/compiler.CompileWithDebugInfo(0xb868c5, 0x6, 0xda1c40, 0xc0001be280, 0xecffd0, 0xf, 0x0, 0x0, 0x0, 0x5158c0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/compiler.go:149 +0xaa
github.com/nspcc-dev/neo-go/pkg/compiler_test.vmAndCompileInterop(0xc000194ea0, 0xbb04dd, 0x79, 0x3ae, 0x5158c0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/vm_test.go:75 +0x10e
github.com/nspcc-dev/neo-go/pkg/compiler_test.vmAndCompile(...)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/vm_test.go:64
github.com/nspcc-dev/neo-go/pkg/compiler_test.eval(0xc000194ea0, 0xbb04dd, 0x79, 0xb7fd00, 0xc00000f6c0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/vm_test.go:42 +0x49
github.com/nspcc-dev/neo-go/pkg/compiler_test.runTestCases.func1(0xc000194ea0)
	/home/alexvanin/Project/Neo/neo-go/pkg/compiler/vm_test.go:31 +0x53
testing.tRunner(0xc000194ea0, 0xc00005f0d0)
	/snap/go/current/src/testing/testing.go:1039 +0xdc
created by testing.(*T).Run
	/snap/go/current/src/testing/testing.go:1090 +0x372
```

--- a/pkg/compiler/codegen.go
+++ b/pkg/compiler/codegen.go
@@ -408,7 +408,12 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 			for _, spec := range n.Specs {
 				vs := spec.(*ast.ValueSpec)
 				for i := range vs.Names {
-					c.constMap[c.getIdentName("", vs.Names[i].Name)] = c.typeAndValueOf(vs.Values[i])
+					info := c.buildInfo.program.Package(c.currPkg.Path())
+					obj := info.Defs[vs.Names[i]]
+					c.constMap[c.getIdentName("", vs.Names[i].Name)] = types.TypeAndValue{
+						Type:  obj.Type(),
+						Value: obj.(*types.Const).Val(),
+					}
 				}
 			}
 			return nil

--- a/pkg/compiler/constant_test.go
+++ b/pkg/compiler/constant_test.go
@@ -72,3 +72,17 @@ func TestGlobalsWithFunctionParams(t *testing.T) {
 	`
 	eval(t, src, []byte("FOO"))
 }
+
+func TestIota(t *testing.T) {
+	src := `package foo
+	const (
+		a = 2 << iota
+		b
+
+		c = 11
+	)
+	func Main() int {
+		return a + b + c
+	}`
+	eval(t, src, big.NewInt(17))
+}

